test/e2e_federation/util.go

```diff
@@ -150,20 +139,85 @@ func createService(clientset *fedclientset.Clientset, namespace, name string) (*
 	return clientset.Services(namespace).Create(service)
 }
 
-func createServiceOrFail(clientset *fedclientset.Clientset, namespace, name string) *v1.Service {
-	service, err := createService(clientset, namespace, name)
+func createService(clientset *fedclientset.Clientset, namespace, name string, clusters fedframework.ClusterSlice) (*v1.Service, error) {
+	if clientset == nil || len(namespace) == 0 {
+		return nil, fmt.Errorf("Internal error: invalid parameters passed to createService: clientset: %v, namespace: %v", clientset, namespace)
+	}
+	By(fmt.Sprintf("Creating federated service %q in namespace %q", name, namespace))
+
+	// Tests can be run in parallel, so we need a different nodePort for
+	// each test.
+	// we add in a array all the "available" ports
+	availablePorts := make([]int32, FederatedSvcNodePortLast-FederatedSvcNodePortFirst)
+	for i := range availablePorts {
+		availablePorts[i] = int32(FederatedSvcNodePortFirst + i)
+	}
+
+	var err error
+	var service *v1.Service
+	// until the availablePort list is not empty, lets try to create the service
+	for len(availablePorts) > 0 {
+		// select the Id of an available port
+		i := rand.Intn(len(availablePorts))
+
+		By(fmt.Sprintf("try creating federated service %q in namespace %q with nodePort %d", name, namespace, availablePorts[i]))
+
+		service, err = createServiceWithNodePort(clientset, namespace, name, availablePorts[i])
+
+		if err == nil {
+			// check if service have been created properly in all clusters.
+			// if the service is not present in one of the clusters, we should cleanup all services
+			err = checkServicesCreation(namespace, name, clusters)
+		}
+
+		if err == nil {
+			// everything was created properly so returns the federated service.
+			return service, nil
+		}
```

Comment:
first`ìf err == nil` is here for validation the the Service was created properly in the Federation-API-Server
the second `ìf err == nil` is here to check that the service have been creation on each federated clusters.

I didn't initially merge both, because I was considering that if `createServiceWithNodePort()` already return an error, we don't need to check on each cluster if the service was created properly.